src/main/groovy/org/jetbrains/gradle/ext/IdeArtifacts.groovy

```diff
@@ -167,12 +168,29 @@ class ArchiveArtifact extends RecursiveArtifact {
       it.buildTo(temp)
     }
 
-    def customArchName = name
-    project.tasks.create(randomName, Zip) {
-      from temp
-      destinationDir destination
-      archiveName customArchName
-    }.execute()
+    createTestArchive(name, temp, destination)
+  }
+
+  def createTestArchive(String archiveName, File inputDir, File destinationDir) {
+
+    ZipOutputStream output = new ZipOutputStream(new FileOutputStream(new File(destinationDir, archiveName)))
+
+    inputDir.eachFileRecurse { file ->
+      if (!file.isFile()) {
+        return
+      }
+
+      String relativePath = inputDir.relativePath(file);
+      output.putNextEntry(new ZipEntry(relativePath));
+      InputStream input = new FileInputStream(file);
```

Comment:
Rather than explicitly opening this, I think you can use `Files.copy(file.toPath(), output)` and have the stream opened and closed properly automatically.

Reply:
👍